analyzers/tests/SonarAnalyzer.UnitTest/TestCases/UseReturnStatement.vb

```diff
@@ -17,37 +19,44 @@
+
     Public Function CaseInsensitive() As Integer
         CASEINSENSITIVE = 42  ' Noncompliant
     End Function
 
     Public Shared Function SharedFunction() As Integer
         SharedFunction = 42  ' Noncompliant
     End Function
-    
+
     Public Function ExplictlyReturnDefaultReturnValue() As Integer
         Return ExplictlyReturnDefaultReturnValue ' Noncompliant {{Do not make use of the implicit return value.}}
     End Function
-        
+
     Public Function ReadAssignementStatements() As Integer
-        dim value as Integer = ReadAssignementStatements ' Noncompliant  {{Do not make use of the implicit return value.}}
-        value += ReadAssignementStatements ' Noncompliant  {{Do not make use of the implicit return value.}}
-        value -= ReadAssignementStatements ' Noncompliant  {{Do not make use of the implicit return value.}}
-        value *= ReadAssignementStatements ' Noncompliant  {{Do not make use of the implicit return value.}}
-        value /= ReadAssignementStatements ' Noncompliant  {{Do not make use of the implicit return value.}}
-        value \= ReadAssignementStatements ' Noncompliant  {{Do not make use of the implicit return value.}}
-        value &= ReadAssignementStatements ' Noncompliant  {{Do not make use of the implicit return value.}}
-        value ^= ReadAssignementStatements ' Noncompliant  {{Do not make use of the implicit return value.}}
+        Dim value As Integer = ReadAssignementStatements ' Noncompliant  {{Do not make use of the implicit return value.}}
+        value += ReadAssignementStatements  ' Noncompliant {{Do not make use of the implicit return value.}}
+        value -= ReadAssignementStatements  ' Noncompliant {{Do not make use of the implicit return value.}}
+        value *= ReadAssignementStatements  ' Noncompliant {{Do not make use of the implicit return value.}}
+        value /= ReadAssignementStatements  ' Noncompliant {{Do not make use of the implicit return value.}}
+        value \= ReadAssignementStatements  ' Noncompliant {{Do not make use of the implicit return value.}}
+        value &= ReadAssignementStatements  ' Noncompliant {{Do not make use of the implicit return value.}}
+        value ^= ReadAssignementStatements  ' Noncompliant {{Do not make use of the implicit return value.}}
         value <<= ReadAssignementStatements ' Noncompliant {{Do not make use of the implicit return value.}}
         value >>= ReadAssignementStatements ' Noncompliant {{Do not make use of the implicit return value.}}
         Return value
     End Function
-    
-     Public Function ImplementedInterfaceMethod() As Integer Implements IInterface.ImplementedInterfaceMethod
+
+    Public Function ImplementedInterfaceMethod() As Integer Implements IInterface.ImplementedInterfaceMethod
         ImplementedInterfaceMethod = 42 ' Noncompliant
     End Function
 
+    Private Function ArgumentName() As Integer
+        WithExplicitArgumentName(Something:=ArgumentName)    ' Noncompliant
```

Comment:
I don't understand the purpose of this test.
- no value is assigned to the function name
- the fact that it's an implicit return value should work with just a function call, right?

Reply:
like
```
Private Function X() As Integer
  FooThatReturnsInteger()
End Function
```

Reply:
Missing `Return` statement is not important here. Important is the reading of `ArgumentName` implicit variable.

Reply:
There's a difference between
```
WithExplicitArgumentName(Something:=ArgumentName)   ' Reading implicit variable that holds the return value
WithExplicitArgumentName(Something:=ArgumentName()) ' Recursive invocation
```